mergify_engine/rules/__init__.py

```diff
@@ -95,35 +160,79 @@ async def eval(self, obj: filter.GetAttrObjectT) -> bool:
         self.match = await filter.Filter({"and": self.conditions})(obj)
         return self.match
 
-    def __getitem__(self, idx: int) -> RuleCondition:
-        return self.conditions[idx]
+    def iter_root_rule_conditions(
+        self,
+    ) -> typing.Iterator[RuleCondition]:
+        for condition in self.conditions:
+            if isinstance(condition, RuleCondition):
+                yield condition
 
-    def __iter__(self) -> typing.Iterator[RuleCondition]:
-        # TODO(sileht): browse the complete tree later
-        return iter(self.conditions)
+    @classmethod
+    def _traverse_rule_condition(
```

Comment:
```suggestion
    def walk(
```
`walk` is more often used in this case like `os.walk`